images/spz/spz/templates/internal/registrations.html

```diff
@@ -22,4 +22,22 @@ <h3 class="ui dividing header">Import</h3>
         <button type="submit" class="fluid ui positive button">Einlesen</button>
     </form>
 </div>
+
+<div class="row">
+    <h3 class="ui dividing header">Matrikelnummer/Kürzel-Überprüfung</h3>
+    <form class="ui form" method="post" action="{{ url_for('registrations_verify') }}" enctype="multipart/form-data">
+        {{ csrf_field() }}
+        {{ render_input(form.tag) }}
+        {% if tag_exists %}
```

Comment:
As far as I know `None` also evaluates to `False` when being "on its own". So when you pass no object at all [here](https://github.com/spz-signup/spz-signup/pull/62/files#diff-c33e317b7d37b8e87cb84f6fdc929f0aR195), the flow will result in the `else` branch and will try to emit `tag` which ... tada, isn't defined. So it will likely emit "Matrikelnummer/Kürzel "None" ist noch nicht in der Datenbank." as a negative message. This is wrong, since the user didn't request any search at all at this point.

My suggestion is: Replace the `if` condition with a more exact one (`if tag_exists == True`) and the `else` branch with an `elif` and an explicit `elif tag_exisits == False`. This avoids that you need yet another if-guard + indicator variable for the entire construction.

I also might be totally wrong with my analysis of the current commit since I didn't test this, but that's my prediction. It might also just horribly blow up without emitting a misleading message :boom: